atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueService.java

```diff
@@ -2129,53 +2126,60 @@ public void compactInternally(TableReference tableRef) {
 
     @Override
     public NodeAvailabilityStatus getNodeAvailabilityStatus() {
-        Map<InetSocketAddress, CassandraClientPoolingContainer> currentPools = clientPool.currentPools;
-        int replicationFactor = configManager.getConfig().replicationFactor();
+        if (!doesConfigReplicationFactorMatchWithCluster()) {
+            return NodeAvailabilityStatus.TERMINAL;
```

Comment:
We should consider what we expect a client to do when they get NodeAvailiabilityStatus.TERMINAL - especially as there are a couple of distinct reasons why we give that phase, and both require specific config changes. Do we want to pass a message along with the `TERMINAL` status?